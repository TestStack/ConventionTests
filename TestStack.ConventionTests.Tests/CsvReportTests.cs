namespace TestStack.ConventionTests.Tests
{
    using System.Linq;
    using ApprovalTests.Reporters;
    using NUnit.Framework;
    using TestAssembly.Collections;
    using TestStack.ConventionTests.ConventionData;
    using TestStack.ConventionTests.Tests.TestConventions;

    [UseReporter(typeof(DiffReporter))]
    public class CsvReportTests
    {
        [Test]
        public void Can_run_convention_with_simple_reporter()
        {
            Convention.IsWithApprovedExeptions(new CollectionsRelationsConvention(), new Types("Entities")
            {
                TypesToVerify =
                    typeof (Leaf).Assembly.GetExportedTypes()
                        .Where(t => t.Namespace == typeof (Leaf).Namespace).ToArray()
            });
        }
    }
}