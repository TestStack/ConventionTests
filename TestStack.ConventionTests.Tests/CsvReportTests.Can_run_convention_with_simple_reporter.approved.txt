collection,item,can add,can remove
TestAssembly.Collections.Branch,TestAssembly.Collections.Leaf,False,False
TestAssembly.Collections.Forest,TestAssembly.Collections.Tree,True,True
TestAssembly.Collections.Tree,TestAssembly.Collections.Branch,True,False
TestAssembly.Collections.Tree,TestAssembly.Collections.Leaf,False,False
